tests/pyccel/test_pyccel.py

```diff
@@ -773,18 +773,20 @@ def test_basic_header():
                                         "scripts/classes/classes_1.py",
                                         "scripts/classes/classes_5.py",
                                         "scripts/classes/generic_methods.py",
+                                        "scripts/classes/classes_2_C.py",
                                         ] )
 @pytest.mark.parametrize( 'language', (
         pytest.param("python", marks = pytest.mark.python),
+        pytest.param("c", marks = pytest.mark.c),
         pytest.param("fortran", marks = pytest.mark.fortran)
     )
 )
 
 def test_classes( test_file , language):
-    if language == "python":
-        pyccel_test(test_file, language=language)
-    else:
+    if language == "fortran":
         pyccel_test(test_file, compile_with_pyccel = False, language=language)
+    elif not (language == "c" and not test_file.endswith("_C.py")):
```

Comment:
I find this condition hard to read. Can I suggest:
```suggestion
    elif language != "c" or test_file.endswith("_C.py")):
```
Does this test pass? I am still confused by the lack of `compile_with_pyccel = False`

Reply:
Fixed